BENCHMARKING.md

```diff
@@ -0,0 +1,178 @@
+  Range (min … max):    95.3 ms … 118.4 ms    30 runs
+ 
+Summary
+  
+  /tmp/wild/ripgrep/6/run-with wild ran
+    1.42 ± 0.09 times faster than /tmp/wild/ripgrep/6/run-with mold
+    9.14 ± 0.46 times faster than /tmp/wild/ripgrep/6/run-with ld
+```
+
+### Run benchmark with poop
+
+An alternative tool to hyperfine, that reports some additional metrics is [`poop`](https://github.com/andrewrk/poop).
+
+Like hyperfine it takes a number of commands and runs each a number of times and gathers statistics about each tune.
+
+Run it using `poop '/tmp/wild/ripgrep/6/run-with ld' '/tmp/wild/ripgrep/6/run-with mold'
+'/tmp/wild/ripgrep/6/run-with wild'`
+
+It should produce output similar to this (with different numbers!):
+
+```text
+Benchmark 1 (6 runs): /tmp/wild/ripgrep/6/run-with ld
+measurement          mean ± σ min …     max outliers delta
+wall_time           911ms ± 20.9ms      897ms … 952ms 0 ( 0%)        0%
+peak_rss            282MB ± 94.3KB      282MB … 282MB 0 ( 0%)        0%
+cpu_cycles          2.24G ± 11.2M       2.22G … 2.25G 0 ( 0%)        0%
+instructions        3.78G ± 8.64K       3.78G … 3.78G 0 ( 0%)        0%
+cache_references    92.2M ± 319K        91.8M … 92.6M 0 ( 0%)        0%
+cache_misses        38.2M ± 428K        37.8M … 38.8M 0 ( 0%)        0%
+branch_misses       9.27M ± 16.7K       9.24M … 9.29M 0 ( 0%)        0%
+
+Benchmark 2 (31 runs): /tmp/wild/ripgrep/6/run-with mold
+measurement          mean ± σ min …     max outliers delta
+wall_time           163ms ± 36.9ms      149ms … 356ms 2 ( 6%)       ⚡- 82.1% ± 3.5%
+peak_rss           7.88MB ± 81.1KB     7.73MB … 8.00MB 0 ( 0%)      ⚡- 97.2% ± 0.0%
+cpu_cycles          1.93G ± 20.9M       1.89G … 1.96G 0 ( 0%)       ⚡- 13.6% ± 0.8%
+instructions        1.95G ± 1.30M       1.95G … 1.95G 1 ( 3%)       ⚡- 48.4% ± 0.0%
+cache_references    43.3M ± 260K        43.0M … 44.3M 1 ( 3%)       ⚡- 53.0% ± 0.3%
+cache_misses        20.6M ± 134K        20.4M … 21.1M 1 ( 3%)       ⚡- 46.1% ± 0.5%
+branch_misses       7.12M ± 42.0K       7.02M … 7.19M 0 ( 0%)       ⚡- 23.2% ± 0.4%
+
+Benchmark 3 (50 runs): /tmp/wild/ripgrep/6/run-with wild
+measurement          mean ± σ min …     max outliers delta
+wall_time           101ms ± 3.10ms     95.2ms … 109ms 1 ( 2%)       ⚡- 88.9% ± 0.7%
+peak_rss            235MB ± 200KB       234MB … 235MB 1 ( 2%)       ⚡- 16.8% ± 0.1%
+cpu_cycles          1.24G ± 15.6M       1.19G … 1.27G 3 ( 6%)       ⚡- 44.4% ± 0.6%
+instructions        1.21G ± 443K        1.21G … 1.21G 0 ( 0%)       ⚡- 67.9% ± 0.0%
+cache_references    34.0M ± 505K        33.1M … 35.0M 0 ( 0%)       ⚡- 63.1% ± 0.5%
+cache_misses        14.2M ± 171K        13.9M … 14.5M 0 ( 0%)       ⚡- 62.9% ± 0.5%
+branch_misses       3.42M ± 11.0K       3.40M … 3.45M 0 ( 0%)       ⚡- 63.1% ± 0.1%
+```
+
+NOTE: `poop` uses the first command as the reference the others are compared against, so if focusing on wild, you might
+want to re-order the commands and invoke `poop` thus:
+
```

Comment:
It might also be worthwhile mentioning that the `peak_rss` for mold and wild will not be at all accurate due to them both forking and that if you're wanting to look at rss. Actually, for `hyperfine`, it's also worth mentioning that the `User` and `System` measurements for wild and mold will be meaningless for the same reason - although I guess the sample results you pasted above are without your fork PR.

Reply:
Fixed